### DISCUSSION TITLE
stuck on verifying

### DISCUSSION THREAD

**Discussion by MightySamuraimax** [RareDevs/Rare#379] [General]
.I have tried downloading games but all the games are stuck on the verifying part .one time fortnite worked but after a update was out its stuck on verifying again .
i request you to give me a solution please.
Thanks

**Comment by loathingKernel** [comment]
You mean in the installation dialog while it verifies your settings before allowing you to continue? Was there any failure? If there was a failure, there should be a message at the bottom of that dialog describing it.

Also, please attach the log file when to issue happened as described here: https://github.com/RareDevs/Rare?tab=readme-ov-file#reporing-issues

**Reply by MightySamuraimax** [reply]
yes,thanks for the reply i will contact you if i deal with any problem but one question. Does verifying always take so long? i think that's like the main problem for me its takes so much time just to verifying like i speeds up to a certain % and stops for a long time and then continues and does the same

**Reply by loathingKernel** [reply]
We are talking about different things here. Let me put it in a simpler way, 

Are we talking about this? 
![image](https://github.com/RareDevs/Rare/assets/142770/ebf5f05c-f69c-40d8-8022-c437bcbe71e4)

or this?
![image](https://github.com/RareDevs/Rare/assets/142770/b7db6cc8-4fdb-4a8d-9edf-b59783c7f92a)

The first one does have a percentage, and the speed of it depends on the game, your OS and your hardware. There is nothing more that Rare can do in that regard as it is tied to what `legendary` does in the background. It sometimes feels like it jumps values because due to `legendary`'s API we do not have accurate information about the process while validating large files, so it looks like it is stuck when it encounters them until it validates the whole file.

The second one depends on the size of the game and the quality of your internet connection.

**Reply by MightySamuraimax** [reply]
oh yea the first one actually the verification gets stuck or takes a long time for example
![image_2024-02-01_193500121](https://github.com/RareDevs/Rare/assets/79087347/d713c173-8181-4495-acad-63b47e05797e)
in this , it has been stuck at 93 % for 20 minutes or even more as now we are speaking and in larger size game like gta it gets stuck for hours. is there anyway i can fix this?


Edit: it moved 1%
![image](https://github.com/RareDevs/Rare/assets/79087347/312c1ab9-4327-4d79-b1e1-0f85f9450ce9)
is my hardware the problem here? or is there any way i can speed it maybe or fix it ig

**Reply by loathingKernel** [reply]
If it completes eventually, meaning Rare doesn't crash or aborts the process, no, there is no way to fix this. This is solely due to hardware and the OS. Take a look at the logs, it will show if anything is crashing, which I don't think is the case here.

**Reply by MightySamuraimax** [reply]
![image](https://github.com/RareDevs/Rare/assets/79087347/1ca2293f-a2b4-4bde-9758-6da0decf44c7)
is this an error?

**Reply by loathingKernel** [reply]
No, it is not. The ERROR here is due to missing information in the Store tab.

**Reply by loathingKernel** [reply]
One thing you could do is exclude the directory where you install games from epic store from your antivirus, so it won't get scanned while a program operates on the files. This should speed up things a bit.

**Reply by MightySamuraimax** [reply]
ok then ig i cant do anything about the speed.Thanks for your help have a great day:)

**Reply by MightySamuraimax** [reply]
problem is i dont have any antivirus lol

**Reply by MightySamuraimax** [reply]
should i try turning off the firewall and verifying?

**Reply by loathingKernel** [reply]
No, firewall shouldn't have an effect on it. If you don't even have Windows Antivirus (Windows Defender), then there isn't much else you could do.

**Reply by MightySamuraimax** [reply]
um srry for the late answer and stuff but i seemed that the games only download in my c drive (ssd) and not in my hdds idk what the problem is but maybe you have a answer.i dont expect a reply but just trying it out

**Reply by loathingKernel** [reply]
You can select the folder when you install a game or set a pre-defined default one in the settings
